src/containers/ProfilePage/ProfilePage.duck.js

```diff
@@ -230,9 +241,20 @@ export const loadData = (params, search, config) => (dispatch, getState, sdk) =>
   // Note 2: In private marketplace mode, this page won't fetch data if the user is unauthorized
   const isAuthorized = currentUser && isUserAuthorized(currentUser);
   const isPrivateMarketplace = config.accessControl.marketplace.private === true;
+  const hasNoViewingRights = currentUser && !hasReadPermission(currentUser);
   const canFetchData = !isPrivateMarketplace || (isPrivateMarketplace && isAuthorized);
+  // On a private marketplace, show current user's own page even if they don't have viewing rights
```

Comment:
Add 'active' state somehow to this sentence.

"show current user's own page for active aka approved users even if they don't have viewing rights"

Reply:
Comment updated!